test/gpu/jit.cpp

```diff
@@ -356,4 +356,50 @@ TEST_CASE(compile_math)
     });
 }
 
+// NOLINTNEXTLINE
+const std::string assert_template = R"__migraphx__(
+#include <migraphx/kernels/math.hpp>
+#include <migraphx/kernels/types.hpp>
+using namespace migraphx;
+extern "C" {
+__global__ void kernel(void*) 
+{
+    static_assert(numeric_max<${type}>() == std::numeric_limits<${type}>::max(), "");
+    static_assert(numeric_lowest<${type}>() == std::numeric_limits<${type}>::lowest(), "");
+}
+}
+
+int main() {}
+
+)__migraphx__";
+
+TEST_CASE(assert_int_type_min_max)
+{
+    std::vector<std::string> data_types;
+    migraphx::gpu::hip_compile_options options;
+    for(auto&& t : migraphx::shape::types())
+    {
+        if(contains({migraphx::shape::bool_type,
+                     migraphx::shape::tuple_type,
+                     migraphx::shape::int64_type,
+                     migraphx::shape::uint64_type},
+                    t))
+            continue;
+        auto name = migraphx::shape::cpp_type(t);
+
+        migraphx::shape::visit(t, [&](auto as) {
+            if(!as.is_integral())
```

Comment:
We should check floating point types as well.

Reply:
I was using `==` to do the comparison between two values, and hip compilation would error out saying that I shouldn't do that.

I'm wondering if there's a kernel utility in migraphx that can tell if the two float point values are close enough?

Reply:
I think it would be better to disable the warning for this test. I think you can do `options.params = "-Wno-float-equal"`.

Reply:
Done in dc852557c2ab6cb6c44e4ac140225c1413906a26